arch/arm/soc/nordic_nrf5/nrf52/Makefile

```diff
@@ -10,6 +10,7 @@ soc-cflags += -DNRF52840_XXAA
 endif
 
 obj-y += soc.o
+obj-$(CONFIG_NRF52_ARM_MPU_ENABLE) += arm_mpu_regions.o
```

Comment:
This might be because it refers to ARM's standard MPU, as opposed to a custom MPU implemented by the IC vendor